src/transformers/pipelines/text2text_generation.py

```diff
@@ -75,7 +76,14 @@ def _sanitize_parameters(
 
         if clean_up_tokenization_spaces is not None:
             postprocess_params["clean_up_tokenization_spaces"] = clean_up_tokenization_spaces
-
+        
+        if stop_sequence is not None:
+            stop_sequence_ids = self.tokenizer.encode(
+                stop_sequence
+            )
+            if len(stop_sequence_ids) > 3:
```

Comment:
Why `3` ?

I think you're looking for `tokenizer.encode(stop_sequence, add_special_tokens=False)` which should raise a warning if its length is `>1` .

Not all tokenizer insert BOS and EOS, that's why we have to do it this way.